packages/nx/src/project-graph/plugins/utils.ts

```diff
@@ -61,11 +66,17 @@ export async function runCreateNodesInParallel(
 
   const promises: Array<Promise<void>> = configFiles.map((file) => {
     performance.mark(`${plugin.name}:createNodes:${file} - start`);
-    // Result is either static or a promise, using Promise.resolve lets us
-    // handle both cases with same logic
-    const value = Promise.resolve(
-      plugin.createNodes[1](file, options, context)
-    );
+
+    const value = new Promise<CreateNodesResult>((resolve, reject) => {
+      try {
+        Promise.resolve(plugin.createNodes[1](file, options, context))
+          .then(resolve)
```

Comment:
This looks wrong to me? Can this be a `try catch finally`?

```

    try {
      const value = await plugin.createNodes[1](file, options, context);
      // Existing behavior is to ignore null results of
      // createNodes function.
      if (value) {
        results.push(value);
      }
    } catch (e) {
      errors.push(
        new CreateNodesError({
          error: e,
          pluginName: plugin.name,
          file,
        })
      );
    } finally {
      performance.mark(`${plugin.name}:createNodes:${file} - end`);
      performance.measure(
        `${plugin.name}:createNodes:${file}`,
        `${plugin.name}:createNodes:${file} - start`,
        `${plugin.name}:createNodes:${file} - end`
      );
    }
```

Also... @AgentEnder can you take a look at this? I don't think `r` is actually a `CreateNodesResultWithContext`. It shows as `any` which is why it's allowed to be pushed but this function actually returns `CreateNodesResult[]`? :thinking:

Reply:
Hmmm, interesting. I agree, it looks like the context got dropped here. We should push something like this instead. `r` is just `CreateNodesResult`. Not sure why ts is typing `r` as any here.
```
{
  ...r,
  pluginName: plugin.name,
  file
}
```